spyder/app/tests/test_mainwindow.py

```diff
@@ -1196,6 +1198,68 @@ def test_varexp_magic_dbg(main_window, qtbot):
     assert shell._control.toHtml().count('img src') == 1
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
+@pytest.mark.skipif(PY2, reason="It times out sometimes")
+def test_tight_layout_option_for_inline_plot(main_window, qtbot):
+    """
+    Test that the option to set bbox_inches to 'tight' or 'None' is
+    working when plotting inline in the IPython console.
+    """
+    # Set the option so that bbox_inches=None.
+    CONF.set('ipython_console', 'pylab/inline/bbox_inches', False)
+
+    # Wait until the window is fully up.
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    client = main_window.ipyconsole.get_current_client()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # Give focus to the widget that's going to receive clicks
+    control = main_window.ipyconsole.get_focus_widget()
+    control.setFocus()
+
+    # Generate a plot with bbox_inches=None.
+    with qtbot.waitSignal(shell.executed):
+        shell.execute('import matplotlib.pyplot as plt\n'
+                      'fig, ax = plt.subplots()\n'
+                      'ax.set_position([0, 0, 1, 1])\n'
+                      'plt.plot(range(10))')
+
+    # Get the image name from the html, fetch the image from the shell, and
+    # then save it to a file.
+    html = shell._control.toHtml()
+    img_name = re.search('''<img src="(.+?)" /></p>''', html).group(1)
+    qimage1 = shell._get_image(img_name)
+    assert isinstance(qimage1, QImage)
+
+    # Change the option so that bbox_inches='tight'.
+    CONF.set('ipython_console', 'pylab/inline/bbox_inches', True)
+
+    # Restart the kernel and wait until it's up again
+    shell._prompt_html = None
+    client.restart_kernel()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # Generate a plot with bbox_inches='tight'.
+    with qtbot.waitSignal(shell.executed):
+        shell.execute('import matplotlib.pyplot as plt\n'
+                      'fig, ax = plt.subplots()\n'
+                      'ax.set_position([0, 0, 1, 1])\n'
+                      'plt.plot(range(10))')
+
+    # Get the image name from the html, fetch the image from the shell, and
+    # then save it to a file.
+    html = shell._control.toHtml()
+    img_name = re.search('''<img src="(.+?)" /></p>''', html).group(1)
+    qimage2 = shell._get_image(img_name)
+    assert isinstance(qimage2, QImage)
+
+    # Compare both images and assert that they are not identical.
+    assert qimage1 != qimage2
```

Comment:
Could you use the Matplotlib function I mentioned before to do a more meaningful comparison?

Reply:
OK I think I understand what were suggesting now. 

I need to produce a sample image locally first and then upload it along with my PR. Than, I need to compare the results of the test with that sample image. I didn't understood what you meant at first, sorry for that. That is why I failed to see the point to use the matplotlib function to compare both images, while I could simply compare the QImages.

Reply:
It is done. It is failing for PyQt4. Probably because the inline output is slightly different than for PyQt5. I'm tempted to skip it.

Reply:
I've just created a new Python 3.5 env. with PyQt4. I've noticed that the version of matplotlib that is installed in this environment is very old: version 1.5.1. So there is noticeable differences between the graphs produces with the default parameters.

I think I can make this work by setting stricter constraints on the plot.

Reply:
Ok no, scrap my last comment. It is set to 'tight' by default and I don't know why it is still failing... !

Reply:
Carlos, the resulting figure when `bbox_inches='tight'` depends on the matplotlib version. Therefore, I cannot produce a unique sample figure that will allow the test to pass on every environment we are supporting for the case where `bbox_inches='tight'`.

I can create a 2 different versions of the sample image that will allow the test to pass with recent and older versions of Matplotlib. However, that will make the test highly dependent on future changes of the matplotlib `tight_layout` mechanics.

A better option imo would be to assert that the resulting figure on the CI fit the sample image only for the `bbox_inches=None` case and assert that the comparison fails when `bbox_inches='tight'` with the same sample figure.

Reply:
> A better option imo would be to assert that the resulting figure on the CI fit the sample image only for the bbox_inches=None case and assert that the comparison fails when bbox_inches='tight' with the same sample figure.

Ok, no problem.

Reply:
Also, if you want to skip this test on PyQt4, please do.

Reply:
Is it possible to generate the sample image in matplotlib but without using Spyder as part of the test, and then compare with the image generated in Spyder's IPython console? That should make sure that the text works with whatever matplotlib version is installed on the test computer (at the cost of requiring more test time).

Reply:
That's a good idea too.

Reply:
@jitseniesen thanks for the idea. This is interesting.

I think what could be done is to save the figure with `savefig` by specifying the `bbox_inches` argument and then compare the result with the figure produced inline. I think that should do it!

Reply:
Oh my gosh... It worked ! 😱 
Thank you @jitseniesen